modnet/preprocessing.py

```diff
@@ -128,43 +144,65 @@ def get_cross_nmi(df_feat: pd.DataFrame, drop_thr: float = 0.2, return_entropy =
     else:
         seed = np.random.RandomState()
 
+    if kwargs.get("n_neighbors"):
+        n_neighbors = kwargs.pop("n_neighbors")
+    else:
+        n_neighbors = 3
+
     # Prepare the output DataFrame and compute the mutual information
     mutual_info = pd.DataFrame([], columns=df_feat.columns, index=df_feat.columns)
 
+    #create pool of workers
+    pool = Pool(processes=cpu_count())
+    LOG.info(f'Multiprocessing on {cpu_count()} workers.')
+
     # Compute the "self" mutual information (i.e. information entropy) of the features
     LOG.info('Computing "self" MI (i.e. information entropy) of features')
     diag = {}
+    tasks = []
```

Comment:
This comment got lost from the last review somehow:


It's hard to see what the changes actually are here, as there's lots of syntax/whitespace changes baked into the PR (see comment above about using an auto-formatter for the whole project). I tried to run a featurization -> feature selection -> fit process based on these changes, and I found that something strange happens in the parallelisation of the cross NMI. It seemed like it was recomputing all the features, or at least the matminer progress bars were being displayed. I'll try to investigate this further tonight

Reply:
This was just me being dumb